x-hooks/use-dao.tsx

```diff
@@ -37,29 +38,37 @@ export function useDao() {
     return isAddress(chain?.registryAddress) && !isZeroAddress(chain?.registryAddress);
   }
 
-  async function disconnect () {
-    const isMetamask = getProviderNameFromConnection(dappkitReact.connection) === WalletProviders.MetaMask;
-    const walletConnector = isMetamask ? metamaskWallet : coinbaseWallet;
-    if (walletConnector?.deactivate)
-      walletConnector?.deactivate();
-    else
-      walletConnector?.resetState();
-    dappkitReact.disconnect();
+  function getLastConnector () {
+    const ids = {
+      "coinbase": "coinbaseWalletSDK",
+      "metamask": "metaMask"
+    };
+    const recentConnectorId = cookies ? cookies["wagmi.recentConnectorId"]?.toLowerCase()?.replace("\"", "") : null;
+    if (!recentConnectorId)
+      return null;
+    if (recentConnectorId === "walletConnect")
+      return connectors?.find(c => (c as any)?.rkDetails?.isWalletConnectModalConnector);
+    return connectors?.find(c => lowerCaseCompare(c?.id, ids[recentConnectorId] || recentConnectorId));
   }
 
-  async function connect () {
-    const lastProvider = window.localStorage.getItem(StorageKeys.lastProviderConnected);
-    const hasProviderMap = !!(window?.ethereum as any)?.providerMap;
-    const selectedProvider =
-      hasProviderMap ? (window?.ethereum as any)?.providerMap?.get(lastProvider) : window.ethereum;
-    if (selectedProvider) {
-      if ((window?.ethereum as any)?.setSelectedProvider)
-        (window.ethereum as any).setSelectedProvider(selectedProvider);
-
-      dappkitReact
-        .setProvider(selectedProvider)
-        .catch(console.debug);
-    }
+  async function connect (): Promise<string | null> {
+    const lastConnector = getLastConnector();

```

Comment:
what happens _if i never connected before_ ?

Reply:
@moshmage this `connect` is only used to auto connect the wallet if a session was already created. The first connect is made by the rainbowkit modal.